Limit number of redis connections and fix ttl bug

anirbanmu opened the pull request:
- Previously (before commit a94719b) each gush object that had a method `client` with a memoized `Client.new` would create a new redis connection. This was causing a lot of redis connections for us & causing us to hit the limit of our redis instance. With commit a94719b, we now hold only open redis connection per thread via a thread local class variable. This lowers the amount of redis connection substantially.
- Commit 1c12ce2 fixes a bug where TTL was not being set correct for job key causing job keys to never be expired.

pokonski commented:
Hey @anirbanmu. Thanks a bunch for the contribution :) Looks like CI will need some fixing due to bundler upgrade.

pokonski commented on lib/gush/client.rb:
```diff
@@ -1,9 +1,22 @@
-require 'connection_pool'
+require 'redis'
+require 'concurrent-ruby'
```
Now that we explicitly use concurrent ruby we should add dependency to .gemspec

pokonski commented on lib/gush/client.rb:
```diff
@@ -1,9 +1,22 @@
-require 'connection_pool'
```
This can be removed as a dependency now

pokonski commented:
General question: instead of reimplemnting cache. Wouldn't it make more sense to cache connection pool?

anirbanmu commented:
> General question: instead of reimplementing connnections cache wouldn't it make more sense to cache connection pool?

Good question. Basically, in my understanding, a connection pool only makes sense if one wants to have a pool of connections that will be shared by multiple consumers. In the design I went with, there is one connection per ruby thread, so there are never multiple consumers for the cached connection. So with the one per thread design, I don't think having a connection pool makes sense.

I think caching a connection pool might make sense, if we changed the design to have a per process connection pool which is then shared by all threads in that process or something like that. That may be a direction worth exploring but I like the cleanliness of not having to worry about a global pool/sharing, and still getting a decrease in # of redis connections.

PS - I fixed travis CI & updated the gemspec as well.

pokonski commented:
@anirbanmu thanks for the detailed explanation. I agree now that Connection Pooling might wait :)
